### PR TITLE
add-savvycal-extension

### PR DESCRIPTION
**Pull request by laulauland**
Hey there! Big fan of Raycast: using it and loving it every day. I couldn't find the extension for Savvycal, another one of my daily drivers, so I built it for myself initially and then figured might as well share it :) 

Props to the Raycast team for a great developer experience!

## Description

Adds a new extension to work with [Savvycal](https://savvycal.com/)

## Screencast

savvycal-1

## Checklist

- [ x ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ x ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ x ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ x ] I checked that files in the `assets` folder are used by the extension itself
- [ x ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/savvycal/src/listSchedulingLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      } catch (error: unknown) {
+        console.error("Error fetching links:\n", error);
+        setLinks({
+          error: error,
+        });
+        showToast({
+          style: Toast.Style.Failure,
+          title: "Savvycal API Error",
+          message:
+            "Check your Savvycal token + username in the extension preferences",
```
Comment (pernielsentikaer) [inline extensions/savvycal/src/listSchedulingLinks.tsx:69]: There is only a token, right?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪

I tested your contribution and it works and looks nice 💪

I have some feedback for you:

- You could use our [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) hook, this might ease the job 🙂

- When wrong credentials are entered I keep seeing the loading, could you return an empty array?
image

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I'm looking forward to testing this extension again 🔥

File: extensions/savvycal/src/listSchedulingLinks.tsx (hunk 1 above)
Comment (keturiosakys) [reply extensions/savvycal/src/listSchedulingLinks.tsx:69]: ah yep, will change this. The initial version asked for a username URL as well

Comment (pernielsentikaer) [reply extensions/savvycal/src/listSchedulingLinks.tsx:69]: Ah, makes sense 🙂

**Comment by keturiosakys** [comment]
Thanks for the comments @pernielsentikaer! Added the requested changes - using useFetch for main function, returning an empty array on failure - as well as refactoring some elements

**Comment by pernielsentikaer** [comment]
Hi @keturiosakys 👋 

I added a few additions, could you please review them and correct accordingly if you find anything I missed 🙂

**Comment by keturiosakys** [comment]
Thank you @pernielsentikaer! These are very useful 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**